codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/RestJsonProtocolGenerator.java

```diff
@@ -44,55 +61,305 @@
     }
 
     @Override
-    protected void generateDocumentBodyShapeSerializers(GenerationContext context, Set<Shape> shapes) {
+    protected void generateOperationDocumentSerializer(
+            GenerationContext context, OperationShape operation
+    ) {
+        Model model = context.getModel();
+
+        HttpBindingIndex bindingIndex = model.getKnowledge(HttpBindingIndex.class);
+
+        Set<MemberShape> bindingMap = bindingIndex.getRequestBindings(operation).values().stream()
```

Comment:
that is not a map

Reply:
`getRequestBindings` has an overloaded method that takes a `Location` constraint. This applies a few places.

Also, it's likely you want to name this after the type of binding you'll receive: `documentBindings`